### PR TITLE
Improvements to the documentation and parsing of dsc files (Apple Unified Logging)

### PR DESCRIPTION
**Pull request by rick-slin**
Added documentation avec version 2 of dsc files.
 Completed the DSCFile class.

### CODE REVIEW

**Comment by rick-slin** [comment]
I hope this is in line with this repo's guidelines. I look forward to your comments.

**Comment by joachimmetz** [comment]
@rick-slin thx, taking a look as soon as time permits.

File: documentation/Apple Unified Logging and Activity Tracing formats.asciidoc [hunk 1]
```diff
@@ -893,30 +894,48 @@ The Shared-Cache Strings (dsc) file header is 16 bytes of size and consists of:
...
-| 4 | 2 | 1 | Format major version
+| 4 | 2 | 1 or 2 | Format major version
 | 6 | 2 | 0 | Format minor version
 | 8 | 4 | | Number of ranges (range count)
 | 12 | 4 | | Number of UUIDs (uuid count)
 |===
 
 === Shared-Cache Strings (dsc) range descriptor
 
+==== Version 1 ====
```
Comment (joachimmetz) [inline documentation/Apple Unified Logging and Activity Tracing formats.asciidoc]: Trailing `====` not needed

File: dtformats/unified_logging.py [hunk 2]
```diff
@@ -24,13 +24,20 @@ class DSCFile(data_format.BinaryDataFile):
       ('number_of_ranges', 'Number of ranges', '_FormatIntegerAsDecimal'),
       ('number_of_uuids', 'Number of UUIDs', '_FormatIntegerAsDecimal')]
 
-  _DEBUG_INFO_RANGE_DESCRIPTOR = [
+  _DEBUG_INFO_RANGE_DESCRIPTOR_V1 = [
       ('uuid_descriptor_index', 'UUID descriptor index',
        '_FormatIntegerAsDecimal'),
-      ('range_offset', 'Range offset', '_FormatIntegerAsHexadecimal8'),
-      ('data_offset', 'Data offset', '_FormatIntegerAsHexadecimal8'),
+      ('range_offset', 'Data offset', '_FormatIntegerAsHexadecimal8'),
```
Comment (joachimmetz) [inline dtformats/unified_logging.py]: Why mix the names and descriptions?

File: dtformats/unified_logging.py [hunk 3]
```diff
@@ -79,43 +86,191 @@ def _ReadFileHeader(self, file_object):
...
+    """Reads the list of ranges.
+
+    Args:
+      file_offset (int): offset of the start of ranges data relative
+          to the start of the file.
+      file_object (file): file-like object.
+      range_descriptors (list): the list of range descriptors.
+
+    Returns:
+      list: list of ranges.
```
Comment (joachimmetz) [inline dtformats/unified_logging.py]: use `tuple[]`

File: dtformats/unified_logging.py [hunk 4]
```diff
@@ -79,43 +86,191 @@ def _ReadFileHeader(self, file_object):
...
+      ParseError: if the file cannot be read.
+    """
+    if version == 1:
+      data_type_map = self._GetDataTypeMap('dsc_range_descriptor_v1')
+
+    elif version == 2:
+      data_type_map = self._GetDataTypeMap('dsc_range_descriptor_v2')
+
+    else:
+      raise 'Unsupported format version: {0:d}.'.format(version)
```
Comment (joachimmetz) [inline dtformats/unified_logging.py]: this does not raise a ParseError exception.

File: dtformats/unified_logging.py [hunk 5]
```diff
@@ -79,43 +86,191 @@ def _ReadFileHeader(self, file_object):
...
+        file_offset, file_object, file_header.major_format_version,
+        file_header.number_of_uuids)
+
+    ranges, file_offset = self._ReadRanges(
+        file_offset, file_object, range_descriptors)
+
+    uuid_paths, file_offset = self._ReadUUIDPaths(
+        file_offset, file_object, file_header.number_of_uuids)
+
+    return {'file_header': file_header,
```
Comment (joachimmetz) [inline dtformats/unified_logging.py]: Does not appear to be used.

File: tests/unified_logging.py [hunk 6]
```diff
@@ -32,7 +165,7 @@ def testReadFileObject(self):
         debug=True, output_writer=output_writer)
 
     test_file_path = self._GetTestFilePath([
-        'uuidtext', 'dsc', '8E21CAB1DCF936B49F85CF860E6F34EC'])
+        'uuidtext', 'dsc', 'dsc-version1'])
```
Comment (joachimmetz) [inline tests/unified_logging.py:169]: Looks like tests with 8E21CAB1DCF936B49F85CF860E6F34EC now break.

**Review by joachimmetz — COMMENTED** [review]
Left some comments, I've made some changes so I will merge the PR and later make changes.

**Pull request merged**